grpc/nodejs/BUILD

```diff
@@ -16,21 +16,50 @@
 #
 
 package(default_visibility = ["//visibility:public"])
-load("@stackb_rules_proto//node:node_grpc_compile.bzl", "node_grpc_compile")
 load("@graknlabs_dependencies//tool/checkstyle:rules.bzl", "checkstyle_test")
+load("@stackb_rules_proto//node:node_grpc_compile.bzl", "node_grpc_compile")
+load("@npm//@bazel/typescript:index.bzl", "ts_project")
+
 
-node_grpc_compile(
+genrule(
     name = "protocol",
-    deps = [
-        "//protobuf:answer-proto",
-        "//protobuf:concept-proto",
-        "//protobuf:database-proto",
-        "//protobuf:grakn-proto",
-        "//protobuf:options-proto",
-        "//protobuf:query-proto",
-        "//protobuf:session-proto",
-        "//protobuf:transaction-proto",
-    ]
+    outs = [
+        "protobuf/answer_pb.d.ts",
+        "protobuf/answer_pb.js",
+        "protobuf/concept_pb.d.ts",
+        "protobuf/concept_pb.js",
+        "protobuf/database_pb.d.ts",
+        "protobuf/database_pb.js",
+        "protobuf/grakn_pb.d.ts",
+        "protobuf/grakn_pb.js",
+        "protobuf/grakn_grpc_pb.js",
+        "protobuf/grakn_grpc_pb.d.ts",
+        "protobuf/options_pb.d.ts",
+        "protobuf/options_pb.js",
+        "protobuf/query_pb.d.ts",
+        "protobuf/query_pb.js",
+        "protobuf/session_pb.d.ts",
+        "protobuf/session_pb.js",
+        "protobuf/transaction_pb.d.ts",
+        "protobuf/transaction_pb.js",
+    ],
+    cmd = "ln -s $(execpath @npm//grpc_tools_node_protoc_ts/bin:protoc-gen-ts).runfiles/npm/node_modules ./node_modules ;\
+        $(execpath @npm//grpc-tools/bin:grpc_tools_node_protoc) \
+            --plugin='protoc-gen-ts=$(rootpath @npm//:node_modules/grpc_tools_node_protoc_ts/bin/protoc-gen-ts)' \
+            --js_out='import_style=es6,binary:./$(@D)/' \
+            --grpc_out='./$(@D)/' \
+            --ts_out='./$(@D)/' \
+            --proto_path=`dirname $(execpath //:WORKSPACE)` \
+            $(execpaths //protobuf:proto-raw-buffers);",
```

Comment:
This command is a bit brave, but divided by line:
- Symlinks the node runfiles required by the typescript plugin into the directory where the protocol compiler can see it
- Runs the node protoc compiler
- Using the protoc-gen-ts plugin to generate typescript type definitions in addition to the javascript definitions
- Outputting the javascript, typescript, and service definitions to the output directory specified by bazel ($(@D))
- Looking in the workspace for the relative paths to the .proto files
- using the .proto files as specified in the proto-raw-buffers filegroup as the inputs

Reply:
This is excellent documentation, nice work! I think we should add it to the code itself, in comments, so this doesn't get lost in the ether.